### PR TITLE
Internal improvement: Use MetaMask's web3-provider-engine package instead of Truffle's fork

### PR DESCRIPTION
**Pull request by eggplantzzz**
**Note** - This will be a breaking change for @truffle/hdwallet-provider.

This PR gets Truffle off of Truffle's fork of web3-provider-engine and back on to MetaMask's package. Eventually we should probably re-architect this package to use newer libraries.

### CODE REVIEW

**Comment by cds-amal** [comment]
Note: this PR fixes #3777 and #4143

File: packages/hdwallet-provider/src/index.ts [hunk 1]
```diff
@@ -7,15 +7,19 @@ import { Transaction, FeeMarketEIP1559Transaction } from "@ethereumjs/tx";
...
-import WebsocketProvider from "@trufflesuite/web3-provider-engine/subproviders/websocket";
+import NonceSubProvider from "web3-provider-engine/subproviders/nonce-tracker";
+// @ts-ignore
+import HookedSubprovider from "web3-provider-engine/subproviders/hooked-wallet";
+// @ts-ignore
+import ProviderSubprovider from "web3-provider-engine/subproviders/provider";
+// @ts-ignore
+import RpcProvider from "web3-provider-engine/subproviders/rpc";
+// @ts-ignore
+import WebsocketProvider from "web3-provider-engine/subproviders/websocket";
```
Comment (cds-amal) [inline packages/hdwallet-provider/src/index.ts:22]: The current subprovider declaration files ([filters.d.ts and co](https://github.com/trufflesuite/truffle/blob/bump-web3-provider-engine/packages/hdwallet-provider/typings/web3-provider-engine/subproviders/filters.d.ts#L1)) should be updated to drop the  **trufflesuite** scope. This should allow removing the _ts-ignore_ directive

**Review by cds-amal — CHANGES_REQUESTED** [review]
I'm excited about this @eggplantzzz ! I noticed a small typescript thing and can't wait to deprecate/archive our **trufflesuite/web3-provider-engine**

**Comment by eggplantzzz** [comment]
It seems like this should maybe be a breaking change for hdwallet-provider since the typing for the `send` and `sendAsync` methods has changed slightly?

**Review by cds-amal — APPROVED** [review]
Looks good!

**Pull request merged**